### ISSUE TITLE
The video track reloads every minute or less when screen sharing is performed from Safari.

### ISSUE THREAD

**Issue by John-Peter-Sofer** [aws/amazon-chime-sdk-js#1250]
Hello,

Here is a bug that we reproduce with our application and your 'browser' demo application.

**First problem:** 

As you can see on the video above (accelerated 10 times) and the attached screenshot, the resolution sent by Safari is the maximum resolution of the MacBook Pro, 2880x1800. 
This is strange because it exceeds the maximum resolution given in the faq here: "The Amazon Chime SDK for JavaScript supports video resolutions up to 1280x720 at 30 frames per second without simulcast, and 15 frames per second with simulcast."  --> https://aws.amazon.com/chime/chime-sdk/faq/

Video: 

> https://user-images.githubusercontent.com/17233837/117060178-ee175f80-ad20-11eb-9e45-68b6d79434d7.mp4

Screenshot:
 

> <img width="1440" alt="screen" src="https://user-images.githubusercontent.com/17233837/117061868-ff616b80-ad22-11eb-9d74-7c11ffd76f81.png">

**Note:** 

Chrome sends 1440x900 when sharing a screen from the same laptop

**Second problem:** 

Since the video has a lot of movement, and the resolution is very high, the frame rate sent drops drastically after about 1 minute.  A new video track is then created. And it starts again.

**To sum up**: 

A user sharing via Safari his Retina screen displaying content containing a lot of movement will see his content strongly degraded to other users.

The capture is made on another PC using Chrome 90.0.4430.93 on MacOs 11.3.1. (I also reproduce this problem on a firefox or Chrome running on Linux).

Platform

OS: Mac OS 11.3.1 -- MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports) Intel
Browser: Safari
Browser Version: Version 14.1 (16611.1.21.161.6)
Amazon Chime SDK GitHub Commit: 7438e2c

**Comment by royliwei** [comment]
Thanks for reporting the issues. 
1. The "video resolutions up to 1280x720 at 30 frames per second without simulcast, and 15 frames per second with simulcast" is for video tile. I can reproduce the issue that safari is sending 2880x1800 during content sharing. 
2. Just want to clarify, when using chrome and firefox, does the frameRate drops below 5fps as well? Does chrome send resolution 1440x900 at 15fps?

**Comment by John-Peter-Sofer** [comment]
2. I meant that I was observing the problem on Chrome or Firefox (as a receiver) when the one sending the screenshare is on Safari. 

Now to answer your question:

- On Google (which shares), it is constant: 1440x900 at 15 fps

- On Firefox (which shares), the resolution adapts from 2880x1800 to 1440x900. FPS = 8.

Attached are the webrtc-internals captures for each browser that sends its screen share:

**Safari**
![from_safari](https://user-images.githubusercontent.com/17233837/118003832-364e0780-b349-11eb-9468-97bf8a17a07f.png)

**Chrome**
![from_chrome](https://user-images.githubusercontent.com/17233837/118003833-36e69e00-b349-11eb-8ce8-ef7b9f54a85b.png)

**Firefox**
![from_firefox](https://user-images.githubusercontent.com/17233837/118003836-377f3480-b349-11eb-82b7-1e633f140d76.png)

**Comment by dchen-aws** [comment]
Sorry for getting back to you late on this one.

On the high level note, we are almost at browsers mercy to produce the video stream for screen capture and different providers may have different implementation. For Safari, the sudden drop in framerate can be caused by a number of different factors, CPU, bandwidth constraint or just encoder not able to keep up. 

The graphs shared before seems to be on the receive side. Can you take a look at the send/encode stats and share that with us?
